src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -138,6 +147,21 @@ class SharedLinkSection extends React.Component<Props, State> {
             }
         }
 
+        if (
+            prevProps.sharedLink.permissionLevel !== '' &&
+            prevProps.sharedLink.permissionLevel !== CAN_EDIT &&
+            sharedLink.permissionLevel === CAN_EDIT
+        ) {
+            this.setState({ isPermissionElevatedToEdit: true });
+        }
+
+        if (
+            isPermissionElevatedToEdit &&
+            (sharedLink.permissionLevel !== CAN_EDIT || sharedLink.accessLevel !== ANYONE_IN_COMPANY)
```

Comment:
We need this check as well to account for the following scenarios:

Given that the current shared link has the following:
accessLevel -> ANYONE_IN_COMPANY
permissionLevel -> CAN_EDIT

If the user then decides to update permissionLevel to be something other than `CAN_EDIT`,
then we should set `isPermissionElevatedToEdit: false`.


If the user then decides to update accessLevel to be something other than `ANYONE_IN_COMPANY`
then we should set `isPermissionElevatedToEdit: false`.

The goal behind this is that it acts as a sort of "cleanup" - we don't want the component to be in a stale state where `isPermissionElevatedToEdit` is still true even though the user has updated either the accessLevel or permissionLevel to other values.